src/Component/src/Exception/UnexpectedTypeException.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Exception;
+
+class UnexpectedTypeException extends \InvalidArgumentException
+{
+    /**
+     * @param mixed $value
+     */
+    public function __construct($value, string $expectedType)
+    {
+        parent::__construct(sprintf(
+            'Expected argument of type "%s", "%s" given.',
+            $expectedType,
+            is_object($value) ? get_class($value) : gettype($value),
```

Comment:
Should I create issue to replace it with `get_debug_type` once 8.1 will be allowed or rector will catch this?

Reply:
I think so :)

Reply:
Turned out `get_debug_type` is available since php 8.0 and we can make use of it already.

https://php.watch/versions/8.0/get_debug_type

![image](https://github.com/Sylius/SyliusResourceBundle/assets/870747/3b951914-5159-4b5e-8634-e1c3d3ffeaa0)

```suggestion
            get_debug_type($value),
```

Reply:
https://github.com/Sylius/SyliusResourceBundle/issues/868